frontend/src/FullPageRoutes/Matching.tsx

```diff
@@ -75,8 +76,23 @@ function MatchControls({
   onReject,
   onReport,
 }: MatchControlsType): JSX.Element {
+  const [showReport, setShowReport] = React.useState<boolean>(false)
+  const [description, setDescription] = React.useState<string>("")
   return (
     <Grid item container direction="row" justify="space-around">
+      <AbstractModal
+        title="Report"
+        description="Optional report message"
+        isModalOpen={showReport}
+        handleConfirm={() => onReport(matchId, description)}
```

Comment:
should the report dialog be closed after confirmed?

Reply:
The controls will re-render after the report is submitted because a new match will be requested. No need to explicitly set the modal state to false.

Reply:
that didn't happen when I tested it

Reply:
Hmm actually looks like it keeps the same state unless there are no more matches